wolfcrypt/src/port/Renesas/renesas_tsip_util.c

```diff
@@ -2453,6 +2339,213 @@ static byte _tls2tsipdef(byte cipher)
+                }
+                break;
+            #endif
+
+            default:
+                ret = BAD_FUNC_ARG;
+                break;
+        }
+        tsip_hw_unlock();
+    }
+    else {
+        WOLFSSL_MSG("mutex locking error");
+    }
+    WOLFSSL_LEAVE("tsipImportPrivateKey", ret);
+    return ret;
+}
+
+#endif /* WOLFSSL_RENESAS_TSIP_TLS */
+
+/*
+ * Import wrapped public key then convert it into TSIP key_index format.
+ * The target key should be set with tsip_use_PublicKey_buffer in advance.
+ * Acceptable key types are:
+ *   TSIP_KEY_TYPE_RSA2048     rsa 2048 bit key
+ *   TSIP_KEY_TYPE_RSA4096     rsa 4096 bit key(Not supported as of now)
+ *   TSIP_KEY_TYPE_ECDSAP256   ecdsa p256r1 key
+ */
+WOLFSSL_LOCAL int tsipImportPublicKey(TsipUserCtx* tuc, int keyType)
+{
+    int          ret = 0;
+    e_tsip_err_t err = TSIP_SUCCESS;
+    uint8_t* provisioning_key = g_user_key_info.encrypted_provisioning_key;
+    uint8_t* iv               = g_user_key_info.iv;
+    uint8_t* encPubKey;
+
+    WOLFSSL_ENTER("tsipImportPublicKey");
+
+    if (tuc == NULL ) {
+        return BAD_FUNC_ARG;
+    }
+
+    encPubKey = tuc->wrappedPublicKey;
+
+    if (encPubKey == NULL || provisioning_key == NULL || iv == NULL) {
+        WOLFSSL_MSG("Missing some key materials used for import" );
+        return CRYPTOCB_UNAVAILABLE;
+    } 
+
+    if (ret == 0) {
+        if (keyType != tuc->wrappedKeyType) {
+            WOLFSSL_MSG("No public key of specified type is set" );
+            return CRYPTOCB_UNAVAILABLE;
+        }
+    }
+
+    if ((ret = tsip_hw_lock()) == 0) {
+        switch(keyType) {
+            
+        #if !defined(NO_RSA)
+            case TSIP_KEY_TYPE_RSA2048:
+            #if defined(WOLFSSL_RENESAS_TSIP_TLS)
+                tuc->ClientRsa2048PubKey_set = 0;
+            #elif defined(WOLFSSL_RENESAS_TSIP_CRYPTONLY)
+                if (tuc->rsa2048pub_keyIdx != NULL) {
+                    XFREE(tuc->rsa2048pub_keyIdx, NULL,
+                                    DYNAMIC_TYPE_RSA_BUFFER);
+                }
+                
+                tuc->rsa2048pub_keyIdx =
+                (tsip_rsa2048_public_key_index_t*)XMALLOC(
+                    sizeof(tsip_rsa2048_public_key_index_t), NULL, 
+                                    DYNAMIC_TYPE_RSA_BUFFER);
+                if (tuc->rsa2048pub_keyIdx == NULL) {
+                    return MEMORY_E;
+                }
+            #endif
+                err = R_TSIP_GenerateRsa2048PublicKeyIndex(
+                                    provisioning_key, iv, (uint8_t*)encPubKey,
+                                #if defined(WOLFSSL_RENESAS_TSIP_TLS)
+                                     &(tuc->Rsa2048PublicKeyIdx)
+                                #elif defined(WOLFSSL_RENESAS_TSIP_CRYPTONLY)
+                                     tuc->rsa2048pub_keyIdx
+                                #endif
+                                    );
+                printf("err 0x%02x\n", err);
```

Comment:
Is this statement for debugging purpose? If so, please remove it.

Reply:
Thanks, removed